### PR TITLE
[tests] Adjust BuildingSameSolutionTwice_ShouldNotRunACToolTwice to work on an M1 machine.

### PR DESCRIPTION
**Pull request by rolfbjarne**
This test verifies that the build log prints:

    "execution started with arguments: actool --errors --warnings --notices --output-format xml1 --output-partial-info-plist "

However, for M1 machines, we don't call actool directly, we go through 'arch'
instead, and this line is printed:

    Tool arch execution started with arguments: -arch arm64e /usr/bin/xcrun actool --errors --warnings --notices --output-format xml1 --output-partial-info-plist ...

So adjust the test to cope with both of these behaviors.

### CODE REVIEW

**Review by spouliot — APPROVED** [review]

File: tests/mmptest/src/MMPTest.cs [hunk 1]
```diff
@@ -688,7 +688,12 @@ public void MM2003 (string opt)
 		[Test]
 		public void BuildingSameSolutionTwice_ShouldNotRunACToolTwice ()
 		{
-			const string actool = " execution started with arguments: actool --errors --warnings --notices --output-format xml1 --output-partial-info-plist ";
+			Func<string, bool> executedActool = (line) => {
+				if (!line.Contains (" execution started with arguments"))
```
Comment (stephen-hawley) [inline tests/mmptest/src/MMPTest.cs:692]: why not `(lint) => line.Contains (" execution started with arguments") && line.Contains ("actool --errors --warnings --notices --output-format xml1 --output-partial-info-plist ");`?

**Review by stephen-hawley — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: tests/mmptest/src/MMPTest.cs (hunk 1 above)
Comment (rolfbjarne) [reply tests/mmptest/src/MMPTest.cs:692]: No reason, except that I wrote this on a tiny laptop with a small screen.

**Review by tj-devel709 — APPROVED** [review]

File: tests/mmptest/src/MMPTest.cs (hunk 1 above)
Comment (stephen-hawley) [reply tests/mmptest/src/MMPTest.cs:692]: Have you tried a smaller font? 😆

**Review by mandel-macaque — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failure is unrelated: https://github.com/xamarin/maccore/issues/2409

**Pull request merged**